neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,298 @@
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using VMArray = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public sealed class NeoToken : Nep5Token<NeoToken.AccountState>
+    {
+        public override string ServiceName => "Neo.Native.Tokens.NEO";
+        public override string Name => "NEO";
+        public override string Symbol => "neo";
+        public override int Decimals => 0;
+        public BigInteger TotalAmount { get; }
+
+        private const byte Prefix_Initialized = 11;
+        private const byte Prefix_Validator = 33;
+        private const byte Prefix_ValidatorsCount = 15;
+
+        private NeoToken()
+        {
+            this.TotalAmount = 100000000 * Factor;
+        }
+
+        protected override StackItem Main(ApplicationEngine engine, string operation, VMArray args)
+        {
+            switch (operation)
+            {
+                case "initialize":
+                    return Initialize(engine, new UInt160(args[0].GetByteArray()));
+                case "unclaimedGas":
+                    return UnclaimedGas(engine, new UInt160(args[0].GetByteArray()), (uint)args[1].GetBigInteger());
+                case "registerValidator":
+                    return RegisterValidator(engine, args[0].GetByteArray());
+                case "vote":
+                    return Vote(engine, new UInt160(args[0].GetByteArray()), ((VMArray)args[1]).Select(p => p.GetByteArray().AsSerializable<ECPoint>()).ToArray());
+                case "getValidators":
+                    return GetValidators(engine).Select(p => (StackItem)p.ToArray()).ToArray();
+                default:
+                    return base.Main(engine, operation, args);
+            }
+        }
```

Comment:
A still think voting should go to another contract. But no need to do this now... let's develop voting contract first.